### ISSUE TITLE
`Open` Flag class

### ISSUE THREAD

**Issue by theo-lubert** [jakeheis/SwiftCLI#39]
Setting the Flag class (and some others I imagine) to `open`, instead of `public`, could enable nice things.
In my case it would allow me to handle a global `--no-color` flag way more nicely:

```swift
import ColorizeSwift

class GlobalFlag: Flag {
	typealias Action = () -> Void
	let flagAction: Action?

	init(_ name: String, usage: String, defaultValue: Bool = false, action: Action? = nil) {
		flagAction = action
		super.init(name, usage: usage, defaultValue: defaultValue)
	}

	override func setOn() {
		flagAction?()
		super.setOn()
	}
}

// Allows this
struct ColorizationOptions: GlobalOptionsSource {
	let ignoreColorization = GlobalFlag("--no-color", usage: "Disable output colorization") {
		String.isColorizationEnabled = false
	}

	static var options: [Option] {
		return [ignoreColorization]
	}
}

GlobalOptions.source(ColorizationOptions.self)
```

**Comment by jakeheis** [comment]
Absolutely, are there any others in particular you think would be useful to have `open`? `Flag`, `Key`, `Parameter`, `OptionalParameter`, `CollectedParameter`, and `OptionalCollectedParameter` are the ones I could see being useful

**Comment by theo-lubert** [comment]
Not at this point, I did not dig deep enough.
If I do later on, I can always open pull requests.

**Comment by jakeheis** [comment]
Fixed in 01ced61117cc6b917c732f5d04e4a90d44dcd0bc, will close when a new release is tagged

**Comment by jakeheis** [comment]
Done in `4.0.1`
